app/queries/keywords_query.rb

```diff
@@ -1,19 +1,56 @@
 # frozen_string_literal: true
 
 class KeywordsQuery
-  def initialize(user)
-    @keywords = user.keywords.select(Keyword.column_names.excluding('html'))
+  QUERY_PARAMS = [:url_pattern, :keyword_pattern, { link_types: [] }].freeze
```

Comment:
We define the permitted params in the controller as it is related to the controller. So I think we can move it into  `app/controllers/keywords_controller.rb`

Reply:
I should say I was confused for this one: by default I wanted to put these params in the controller. But they are tightly coupled with this query object: When working in this query object, I need to know what params are available in the filters. Also, if I want to add a new filter param, I do not want to edit 2x different controllers (API + Backend) which is not DRY but also risky. 
That is the reason why I created this CONST here and permit the params still in the controller (as permitting params is definitely the controller responsibility). 

Maybe I should just rename them `FILTER_PARAMS' ? :) Let me know @junan

Reply:
Update: Moved into `Filter:: QUERY_PARAMS`